examples/barotropic_instability.jl

```diff
@@ -0,0 +1,288 @@
+# # Barotropic instability in a channel on a β-plane
+#
+# This example calculates the growth rate of the "barotropic" (that is
+# vertically-uniform) instability of a boundary-hugging jet on a ``β``-plane.
+#
+# # The "power method" for calculating growth rates and eigenmodes
+#
+# An "instability" is a small-amplitude solution that develops due to the presence
+# of an unstable "basic state" or background flow. Because these solutions are
+# small amplitude, we express any of the prognostic variables as
+#
+# ```math
+# u = û exp(σ t)
+# ```
+#
+# where ``σ`` is the "growth rate" of the instability. Our object is to
+# compute ``σ`` and also to get a feel for ``û``, which represents the 
+# "eigenmode", or the spatial structure of the instability.
+#
+# The power method iteratively simulates the growth of the instability, using
+# a "rescaling" method to successively isolate the instability from other motions
+# that develop during a simulation of the fully nonlinear equations.
+#
+# For this we design a criterion for stopping a simulation based on the amplitude
+# of the perturbation ``u``-velocity field:
+
+# 
+# # The domain 
+
+using Oceananigans
+
+grid = RegularCartesianGrid(size=(64, 64, 1), x=(-2π, 2π), y=(-2, 5), z=(0, 1),
+                                  topology=(Periodic, Bounded, Bounded))
+
+# # The background flow
+#
+# Our background flow is a boundary-hugging "Bickley jet":
+
+using Oceananigans.Fields
+
+bickley_jet(x, y, z, t) = sech(y)^2
+
+U = BackgroundField(bickley_jet)
+
+# which looks like
+
+using Oceananigans.Grids
+using Plots
+
+y = ynodes(Cell, grid)
+
+background_flow = plot(bickley_jet.(0, y, 0, 0), y, label=nothing,
+                       linewidth=2, xlabel="U(y)", ylabel="y",
+                       title="Boundary-hugging Bickley jet")
```

Comment:
you don't think we should start with the simple symmetric one? I was thinking to try to reproduce a published result (if I can find any)

Reply:
It depends on our goal. If the purpose of the experiment is to illustrate the accuracy of Oceananigans that is the right approach.

My motivation for choosing something non-standard (you might say, my motivation for doing some shenanigans) is to make the example fun and illustrate the ease with which one can do new problems and use their creativity. This example yields a new result that's not in the literature --- a feat that may seem daunting at first. Partly my goal is to illustrate that producing this kind of science is open to everyone.

I think a motivated user could use this script to check that a result matches something published.

Reply:
I accept your view point! :)